File: generator/src/test/resources/swift/expected/RequestUriParamsTest/test-inherited-uri-parameter-generation.output.swift
import PotentCodables
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest(
    obj: [String : Any],
    str: String,
    def: String,
    int: Int
  ) throws -> Sunday.Operation<Empty, [String : AnyValue], TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests/{obj}/{str}/{int}/{def}",
        pathParameters: [
          "obj": try ParameterValues.encode(obj),
          "str": try ParameterValues.encode(str),
          "def": try ParameterValues.encode(def),
          "int": try ParameterValues.encode(int)
        ],
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestUriParamsTest/test-inherited-uri-parameter-generation.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import kotlin.Any
import kotlin.Int
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public suspend fun fetchTest(
    obj: Map<String, Any>,
    str: String,
    def: String,
    int: Int,
  ): Map<String, Any> = this.requestFactory
    .result(
      method = Method.Get,
      pathTemplate = "/tests/{obj}/{str}/{int}/{def}",
      pathParameters = mapOf(
        "obj" to obj,
        "str" to str,
        "def" to def,
        "int" to int
      ),
      acceptTypes = this.defaultAcceptTypes
    )
}
